### PR TITLE
CI: Perform basic checks on PR commits.

### LINKED ISSUE #872: Create CI pre-checks script to check git commits
Create a script that performs basic checks on the new commits to be merged.

Script can check for things like a `Fixes #XXX` tag, subsystem in title, *etc*.

Script will be called as part of the CI workflow crucially before any setup/build/tests run.

See #856.

### PR DESCRIPTION
**Pull request by jodh-intel**
Use the new "checkcommits" program in the "tests" repository
to perform basic checks on PR commits, such as:

- Subject
  - Contains a subsystem.
  - Length is within prescribed limits.

- Body
  - Contains a "Fixes #XXX" comment.
  - Line lengths are within prescribed limits.

Unlike the original implementation (#873), "checkcommits" should
be more robust.

Fixes #872.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @jcvenegas - here it is. Feel free to try and break it! :smile:

**Comment by jodh-intel** [comment]
Boom! `EPERM` ;)

**Comment by chavafg** [comment]
qa-failed

**Comment by chavafg** [comment]
qa-passed

**Comment by jcvenegas** [comment]
Tested it works with my name! :D 
@jodh-intel I see now all the logic it is part of clearcontainers/test. I added to a commit a url:
```
ERROR: commit c3ae74d4494e11f7f44b3ce2b48eda77475ae33b: body line 15 too long (max 72, got 90): "https://github.com/clearcontainers/tests/blob/master/cmd/checkcommits/checkcommits_test.go"
```
I dont know if we would like to add that case in checkcommits_test.go , if the line is too log but there is only one word ignore it.  But that has to go  in the test repo.

Other than the url case, lgtm

**Comment by jcvenegas** [comment]
lgtm

File: .ci/ci-pre-checks.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (dlespiau) [inline .ci/ci-pre-checks.sh]: Idea: Fold the logic inside that bash script into the go binary so we don't have to carry around anything else than checkcommits if/when we add this to other repos.

Comment (jodh-intel) [inline .ci/ci-pre-checks.sh]: Good idea - let me go and add that...

**Comment by jodh-intel** [comment]
Blocked on https://github.com/clearcontainers/tests/pull/23.

**Comment by jodh-intel** [comment]
branch updated.

**Comment by chavafg** [comment]
qa-passed

**Comment by jodh-intel** [comment]
I'm seeing strange Travis behaviour which sometimes causes `checkcommits` to fail mysteriously. I've raised a bug on Travis which explains the details:

- https://github.com/travis-ci/travis-ci/issues/7830

**Comment by jodh-intel** [comment]
Either I borked something or Travis did, so closing this PR and re-opening to try to get a "fresh" Travis environment...

**Pull request closed without merging**